src/earn/EarnEnterAmount.tsx

```diff
@@ -282,45 +325,68 @@ function EarnEnterAmount({ route }: Props) {
     // eventually we may want to do something smarter here, like subtracting gas fees from the max amount if
     // this is a gas-paying token. for now, we are just showing a warning to the user prompting them to lower the amount
     // if there is not enough for gas
-    setTokenAmountInput(token.balance.toFormat({ decimalSeparator }))
+    if (isWithdrawal) {
+      setTokenAmountInput(
+        transactionToken.balance.multipliedBy(pool.pricePerShare[0]).toFormat({ decimalSeparator })
+      )
+    } else {
+      setTokenAmountInput(inputToken.balance.toFormat({ decimalSeparator }))
+    }
     setEnteredIn('token')
+    setMaxPressed(true)
     tokenAmountInputRef.current?.blur()
     localAmountInputRef.current?.blur()
     AppAnalytics.track(SendEvents.max_pressed, {
-      tokenId: token.tokenId,
-      tokenAddress: token.address,
-      networkId: token.networkId,
+      tokenId: transactionToken.tokenId,
+      tokenAddress: transactionToken.address,
+      networkId: transactionToken.networkId,
     })
   }
 
   const onPressContinue = () => {
-    if (!tokenAmount || !depositToken) {
+    if (!tokenAmount || !transactionToken) {
       // should never happen
       return
     }
     AppAnalytics.track(EarnEvents.earn_enter_amount_continue_press, {
-      amountInUsd: tokenAmount.multipliedBy(token.priceUsd ?? 0).toFixed(2),
+      amountInUsd: tokenAmount.multipliedBy(transactionToken.priceUsd ?? 0).toFixed(2),
```

Comment:
isn't `tokenAmount` in units of the `depositToken` for withdraw? Shouldn't there be a conversion?

Reply:
Fixed in d6a16162e1dcf29edc25efbe97dc099f38d8f0b7!